manifests/git_repo.pp

```diff
@@ -0,0 +1,22 @@
+# Setup and configure a git repo
+define katello_devel::git_repo(
+  $source,
+  $github_username = undef,
+) {
+
+  validate_string($source)
+
+  $sources = {origin => "https://github.com/${source}.git"}
```

Comment:
Do most devs do it this way? 

My origin is usually my fork and the upstream remote is called `remote`.